lib/appsignal/utils/argument_cleaner.ex

```diff
@@ -1,17 +1,124 @@
+defmodule Appsignal.Utils.Literal do
+  defstruct [:value]
+end
+
+defimpl Inspect, for: Appsignal.Utils.Literal do
+  def inspect(%Appsignal.Utils.Literal{value: value}, _opts), do: value
+end
+
 defmodule Appsignal.Utils.ArgumentCleaner do
-  alias Appsignal.Utils.Type
+  @maximum_iterable_count 4
+  @maximum_recursion 2
 
-  def clean(argument)
-      when is_boolean(argument) or is_integer(argument) or is_float(argument) or is_pid(argument) or
-             is_port(argument) or is_reference(argument) do
-    argument
-  end
+  def clean_literal(argument), do: %Appsignal.Utils.Literal{value: clean(argument)}
+  def clean(argument), do: clean(argument, @maximum_recursion)
 
-  def clean(argument) when is_map(argument) do
+  def clean(argument, _) when is_nil(argument), do: inspect(argument)
+  def clean(argument, _) when is_atom(argument), do: inspect(argument)
+  def clean(argument, _) when is_binary(argument), do: "\"...\""
+  def clean(argument, _) when is_bitstring(argument), do: "<<...>>"
+
+  def clean(argument, recurse) when is_map(argument) do
     {struct, map} = Map.pop(argument, :__struct__)
 
-    "%#{Appsignal.Utils.module_name(struct)}{#{Enum.map_join(map, ", ", fn {key, value} -> "#{inspect(key)} => #{inspect(clean(value))}" end)}}"
+    contents = if Enum.empty?(map), do: "", else: do_clean_map(map, recurse, struct)
+
+    if struct do
+      "%#{Appsignal.Utils.module_name(struct)}{#{contents}}"
+    else
+      "%{#{contents}}"
+    end
+  end
+
+  def clean(argument, _) when is_integer(argument), do: "integer()"
+
+  def clean(argument, _) when is_float(argument), do: "float()"
+
+  def clean(argument, _) when is_pid(argument), do: "#PID<...>"
```

Comment:
I don't see this show up. I get `"..."` for pids.

```elixir
  def backtrace_error(_conn, _params) do
    port = Port.open({:spawn, "echo hello"}, [:binary])
    backtrace_error_inner(%{pid: System.pid(), nil: nil, true: true, false: false, int: 123, float: 12.3, abc: "def", ghi: %{jkl: %{mno: "pqr", int: 123, float: 12.3}}}, System.pid(), port, nil, true, false, 123, 12.3, "abc")
  end

  def backtrace_error_inner(%{abc: "def", ghi: %{jkl: %{other: v}}} = arg) do
  end

  def backtrace_error_inner(%{abc: "def", ghi: %{jkl: %{other: v}}} = arg, _pid, _port, _nil, _true, _false, _int, _float, _string) do
  end
```

![image](https://github.com/appsignal/appsignal-elixir/assets/282402/fd675bf3-29c6-4fb1-8690-284f1a3aa4cb)

Reply:
The PID data type in Elixir represents the IDs of BEAM "processes" (which are not OS processes, but internal to the VM) -- you would use `self()` to obtain the PID of the current process.

By contrast, `System.pid()` returns the process ID of the OS process that runs the whole instance of the VM. And for some reason, it returns it as a string -- hence why it's omitted as `"..."`.